packages/core/src/scheduler/PropagateTask.ts

```diff
@@ -0,0 +1,57 @@
+/** @license MIT License (c) copyright 2010-2016 original author or authors */
+/** @author Brian Cavalier */
+/** @author John Hann */
+
+import fatal from '../fatalError'
+import { Sink, Time, Task } from '@most/types'
+
+export type PropagateTaskRun<A, B = A> =
+  (time: number, value: A, sink: Sink<B>) => any
```

Comment:
Shouldn’t this be: `(time: number, value: A, sink: Sink<B>) => void`?

I’d really have liked to see it like this:

```ts
export type PropagateTaskRun =
  <A, B = A>(time: number, value: A, sink: Sink<B>) => void
```

But TS has issues. See this SO question, comments and answers: https://stackoverflow.com/questions/56675333/extending-a-union-type-exactly

Reply:
I changed the return type to void but not sure if it's possible to make type arguments to be inferred